magicgui/widgets/_function_gui.py

```diff
@@ -344,3 +366,35 @@ def _docstring_to_html(docs: str) -> str:
     short = f"<p>{ds.short_description}</p>" if ds.short_description else ""
     long = f"<p>{ds.long_description}</p>" if ds.long_description else ""
     return re.sub(r"``?([^`]+)``?", r"<code>\1</code>", f"{short}{long}{params}")
+
+
+_UNSET = object()
+
+
+@contextmanager
+def _function_name_pointing_to_widget(function_gui: FunctionGui):
+    """Context in which the name of the function points to the function_gui instance.
+
+    When calling the function provided to FunctionGui, we make sure that the name
+    of the function points to the FunctionGui object itself (this object).
+    In standard ``@magicgui`` usage, this will have been the case anyway.
+    Doing this here allows @magic_factories to *also* be self-referential.
```

Comment:
Maybe link to the various comments/conversations about this? Or add a minimal example of self-referencing here?

Reply:
expanded comment with example